src/Core/ScreenSpaceError.js

```diff
@@ -0,0 +1,96 @@
+import * as THREE from 'three';
+
+const v = new THREE.Vector3();
+const m = new THREE.Matrix4();
+const m2 = new THREE.Matrix4();
+const m3 = new THREE.Matrix4();
+const basis = [
+    new THREE.Vector3(1, 0, 0),
+    new THREE.Vector3(0, 1, 0),
+    new THREE.Vector3(0, 0, 1),
+];
+
+function computeVectorSizeAtDistance(vector, matrix, camera, distance) {
+    basis[0].set(vector.x, 0, 0);
+    basis[1].set(0, vector.y, 0);
+    basis[2].set(0, 0, vector.z);
+
+    m2.identity();
+    m2.extractRotation(matrix);
+    m3.identity();
+    m3.extractRotation(camera.camera3D.matrixWorldInverse);
+
+    for (let i = 0; i < 3; i++) {
+        const b = basis[i];
+
+        // Apply rotation
+        b.applyMatrix4(m2);
+        // Apply inverse camera rotation
+        b.applyMatrix4(m3);
+        // Move at 'distance' from camera
+        b.z += -distance;
+        // Project on screen
+        b.applyMatrix4(camera.camera3D.projectionMatrix);
+        // cancel z component
+        b.z = 0;
+        b.x = b.x * camera.width * 0.5;
+        b.y = b.y * camera.height * 0.5;
+    }
+
+    return basis.map(b => b.length());
```

Comment:
this looks like an approximation of some distance, which computation is optimized by considering axes independently. could you phrase mathematically the exact distance you are approximating ?